### PR TITLE
feat: Handle semver comparisons

### PR DESCRIPTION
**Pull request by PatrykPiwowarczyk**

### CODE REVIEW

**Review by P-Jag — APPROVED** [review]

File: Sources/FeaturevisorSDK/Conditions.swift [hunk 1]
```diff
@@ -1,6 +1,51 @@
 import FeaturevisorTypes
 import Foundation
 
+struct Semver: Equatable {
```
Comment (polok) [inline Sources/FeaturevisorSDK/Conditions.swift:4]: Please, move to separate file

File: Tests/FeaturevisorSDKTests/ConditionsTests.swift [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+            operator: .equals,
+            value: .boolean(true)
+        )
+        var context = Context()
+        context["isTrue"] = .boolean(true)
+
+        XCTAssertTrue(conditionIsMatched(condition: condition, context: context))
+    }
+
+    func testBooleanNotEquals() {
```
Comment (polok) [inline Tests/FeaturevisorSDKTests/ConditionsTests.swift:20]: Follow the same pattern that we have in other tests. Point out sections like GIVEN, WHEN, THEN

File: Sources/FeaturevisorSDK/Conditions+Semver.swift [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import FeaturevisorTypes
+import Foundation
+
+struct Semver: Equatable {
```
Comment (polok) [inline Sources/FeaturevisorSDK/Conditions+Semver.swift:4]: Just Semver.swift, not Conditions+Semver.swift

File: Sources/FeaturevisorSDK/Instance+Refresh.swift [hunk 4]
```diff
@@ -21,7 +21,8 @@ extension FeaturevisorInstance {
 
         try? fetchDatafileContent(
             from: datafileUrl,
-            handleDatafileFetch: handleDatafileFetch) { [weak self] result in
+            handleDatafileFetch: handleDatafileFetch
```
Comment (polok) [inline Sources/FeaturevisorSDK/Instance+Refresh.swift:24]: Please, double check that these format changes follow the current swift-format config

**Review by polok — APPROVED** [review]

File: Sources/FeaturevisorSDK/Instance+Refresh.swift (hunk 4 above)
Comment (PatrykPiwowarczyk) [reply Sources/FeaturevisorSDK/Instance+Refresh.swift]: Updated the branch from the main, this change disappeared on its own + swiftformating shows no errors

File: Tests/FeaturevisorSDKTests/ConditionsTests+Semver.swift [hunk 5]
```diff
@@ -0,0 +1,89 @@
+import XCTest
+
+@testable import FeaturevisorSDK
```
Comment (polok) [inline Tests/FeaturevisorSDKTests/ConditionsTests+Semver.swift:3]: Last thing. This should be called `SemverTests`

Comment (polok) [reply Tests/FeaturevisorSDKTests/ConditionsTests+Semver.swift:3]: I mean the file name

**Pull request merged**